megatron/training.py

```diff
@@ -461,13 +479,40 @@ def setup_model_and_optimizer(model_provider_func, teacher=False):
     if args.deepspeed:
         print_rank_0("DeepSpeed is enabled.")
         pp = mpu.get_pipeline_model_parallel_world_size()
-        model, optimizer, _, lr_scheduler = deepspeed.initialize(
-            model=model[0],
-            optimizer=optimizer,
-            args=args,
-            lr_scheduler=lr_scheduler,
-            mpu=mpu if args.no_pipeline_parallel else None
-        )
+        if args.data_efficiency_curriculum_learning and build_train_valid_test_datasets_provider is not None:
+            train_ds = None
+            if mpu.get_tensor_model_parallel_rank() == 0:
```

Comment:
We check mpu rank here because we would like only one process on one node to build the datasets? Would this cause issue if the model parallelism group is across multiple nodes? Say some nodes may not get their datasets properly setup?

Reply:
Added following comment: Only need to build dataset on tp rank 0 since Megatron has the broadcast_data() function that broadcast data from tp rank 0.

Reply:
Resolved.